astropy/units/core.py

```diff
@@ -2603,32 +2621,33 @@ def def_unit(
 
             {'latex': r'\\Omega'}
 
-    prefixes : bool or list, optional
+    prefixes : bool or iterable of UnitPrefix, optional
         When `True`, generate all of the SI prefixed versions of the
         unit as well.  For example, for a given unit ``m``, will
-        generate ``mm``, ``cm``, ``km``, etc.  When a list, it is a list of
-        prefix definitions of the form:
+        generate ``mm``, ``cm``, ``km``, etc.  If only a few prefixed
+        versions should be created then an iterable of `UnitPrefix`
+        instances can be specified instead. Default is `False`, which
+        means no prefixed versions will be generated.
 
-            (short_names, long_tables, factor)
+        This function always returns the base unit object, even if
+        multiple scaled versions of the unit were created.
 
-        Default is `False`.  This function always returns the base
-        unit object, even if multiple scaled versions of the unit were
-        created.
-
-    exclude_prefixes : list of str, optional
+    exclude_prefixes : collection of str, optional
         If any of the SI prefixes need to be excluded, they may be
-        listed here.  For example, ``Pa`` can be interpreted either as
-        "petaannum" or "Pascal".  Therefore, when defining the
-        prefixes for ``a``, ``exclude_prefixes`` should be set to
-        ``["P"]``.
+        listed here.  For example, when defining the prefixes for ``a``,
+        ``exclude_prefixes`` should be set to ``["P"]`` so that ``Pa``
+        would still refer to the pascal.
+
+        If a bare `str` is used then the prefixes that will be excluded are
+        the substrings of the `str`, not just its individual characters.
 
     namespace : dict, optional
         When provided, inject the unit (and all of its aliases and
         prefixes), into the given namespace dictionary.
 
     Returns
     -------
-    unit : `~astropy.units.UnitBase`
+    unit : `~astropy.units.NamedUnit`
```

Comment:
The annotation are more specific (thanks to the overloads), but this is still correct.